src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/EndpointContracts/IVSDidSaveTextDocumentEndpoint.cs

```diff
@@ -1,14 +1,14 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT license. See License.txt in the project root for license information.
 
+using Microsoft.CommonLanguageServerProtocol.Framework;
 using Microsoft.VisualStudio.LanguageServer.Protocol;
-using OmniSharp.Extensions.JsonRpc;
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer.EndpointContracts
 {
-    [Serial, Method(Methods.TextDocumentDidSaveName)]
-    internal interface IVSDidSaveTextDocumentEndpoint : IJsonRpcNotificationHandler<DidSaveTextDocumentParamsBridge>,
-        IRegistrationExtension
```

Comment:
The removal of theese `IRegistrationExtension` attributes is suprising. Does CLaSP require support for these?

Reply:
Endpoints for DidSaveTextDocument, DidOpenTextDocument, DidCloseTextDocument, and DidChangeTextDocument used to live on the same class and occupy the same Capabilities object. ClaSP does not (currently) support the same class being registered for multiple endpoints so I needed to split them up. Rather than figure out some system by which they each returned partial parts of the object which the others will edit I picked one to do the registering for all of them and let the rest off the hook.

Reply:
Fine for now, but I'd prefer to clean it up eventually. eg right now we don't register for DidSave, but have a handler.. its odd